### ISSUE TITLE
Sliders inside of a div with data-role="fieldcontain" do not render properly.

### ISSUE THREAD

**Issue by nvahalik** [opitzconsulting/jquery-mobile-angular-adapter#104]
The `<div>` element that jqm creates gets rendered with `display:inline-block`.  So when you wrap jqm's generated markup with your new div: 

```
<div ng-create="...">
    .... slider code ....
</div>
```

That div gets rendered with `display: block` and it breaks the rendering of the form.  Like this:

![Screenshot_1_5_13_11_50_AM](https://f.cloud.github.com/assets/632120/45253/7b8b77c2-5760-11e2-8194-8f8936ad2982.png)

As a workaround, I've added this line to my CSS: 

```
.ui-field-contain div[ngm-create] { display: inline-block; }
```

With that line, the problem disappears:

![Screenshot_1_5_13_11_52_AM](https://f.cloud.github.com/assets/632120/45255/9fd74e08-5760-11e2-8d9c-c5c0168d173b.png)

JSFiddle: http://jsfiddle.net/nvahalik/HhXvC/

The CSS fix is the first line in the CSS frame.

**Comment by nvahalik** [comment]
Consequently, other elements suffer from the "fix".  Grouped checkboxes must be updated as well:

```
.ui-field-contain div[ngm-create].ui-checkbox { display: block; }
```

The JSFiddle has been updated accordingly.

**Comment by tbosch** [comment]
Hi,
thanks for reporting this!
I see that jqm does not insert this div, but the adapter does, you are right!

Tobias

**Comment by tbosch** [comment]
Hi, 
sorry, but the wrapper DIV is required for angular to work, as the angular compiler does not like elements appending siblings...

However, the adapter now adds `display:inline-block` to the wrapper elements, just as you selects. See version 1.2.1-SNAPSHOT.js and this fiddle: http://jsfiddle.net/HhXvC/5/

Tobias

**Comment by tbosch** [comment]
Does this work for you?

**Comment by rodcloutier** [comment]
Works for me. Thanks!

**Comment by rodcloutier** [comment]
It seems we have a similar problem with range input when in a field-contain. 
See fiddle: http://jsfiddle.net/rodcloutier/MhFF6/3/

I think It should be `inline` instead of `inline-block`.

**Comment by tbosch** [comment]
Hi,
this problem will be easily fixed in jqm 1.3, as there all inputs are wrapped by jqm into divs (including ranges). 
Tagging this issue for jqm 1.3 as it will be very easy to solve there.

Tobias

**Comment by tbosch** [comment]
Hi,
I found a way to solve this, see here :http://jsfiddle.net/HhXvC/13/ and the 1.2.1-SNAPSHOT.js
However, there is the restriction now (until jqm 1.3) that ng-repeat or other conditional directives are not allowed directly on sliders, as jqm internally creates two sibling elements without a parent (which it does in jqm 1.3).

Tobias
